triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractTriggerTestWithCanton.scala

```diff
@@ -0,0 +1,240 @@
+
+  // TODO: remove debugging cruft
+  override protected val cantonFixtureDebugMode = true
+
+  protected def toHighLevelResult(s: SValue) = s match {
+    case SRecord(_, _, values) if values.size == 6 =>
+      AbstractTriggerTestWithCanton.HighLevelResult(
+        values.get(0),
+        values.get(1),
+        values.get(2),
+        values.get(3),
+        values.get(4),
+        values.get(5),
+      )
+    case _ => throw new IllegalArgumentException(s"Expected record with 6 fields but got $s")
+  }
+
+  protected val applicationId: ApplicationId = RunnerConfig.DefaultApplicationId
+
+  protected def ledgerClientConfiguration: LedgerClientConfiguration =
+    LedgerClientConfiguration(
+      applicationId = ApplicationId.unwrap(applicationId),
+      ledgerIdRequirement = LedgerIdRequirement.none,
+      commandClient = CommandClientConfiguration.default,
+      token = None,
+    )
+
+  protected def ledgerClientChannelConfiguration: LedgerClientChannelConfiguration =
+    LedgerClientChannelConfiguration.InsecureDefaults
+
+  protected def triggerRunnerConfiguration: TriggerRunnerConfig = DefaultTriggerRunnerConfig
+
+  protected val CompiledDar(packageId, compiledPackages) =
+    readDar(darFile.toPath, speedy.Compiler.Config.Dev)
+
+  protected def getRunner(
+      client: LedgerClient,
+      name: QualifiedName,
+      party: String,
+      readAs: Set[String] = Set.empty,
+  ): Runner = {
+    val triggerId = Identifier(packageId, name)
+
+    Trigger.newTriggerLogContext(
+      triggerId,
+      Party(party),
+      Party.subst(readAs),
+      "test-trigger",
+      ApplicationId("test-trigger-app"),
+    ) { implicit triggerContext: TriggerLogContext =>
+      val trigger = Trigger.fromIdentifier(compiledPackages, triggerId).toOption.get
+
+      Runner(
+        compiledPackages,
+        trigger,
+        triggerRunnerConfiguration,
+        client,
+        timeProviderType,
+        applicationId,
+        TriggerParties(
+          actAs = Party(party),
+          readAs = Party.subst(readAs),
+        ),
+      )
+    }
+  }
+
+  protected def allocateParty(client: LedgerClient)(implicit ec: ExecutionContext): Future[String] =
+    client.partyManagementClient.allocateParty(None, None).map(_.party)
+
+  protected def create(client: LedgerClient, party: String, cmd: CreateCommand)(implicit
+      ec: ExecutionContext
+  ): Future[String] = {
+    val commands = Seq(Command().withCreate(cmd))
+    val request = SubmitAndWaitRequest(
+      Some(
+        Commands(
+          party = party,
+          commands = commands,
+          ledgerId = client.ledgerId.unwrap,
+          applicationId = ApplicationId.unwrap(applicationId),
+          commandId = UUID.randomUUID.toString,
+        )
+      )
+    )
+    for {
+      response <- client.commandServiceClient.submitAndWaitForTransaction(request)
+    } yield response.getTransaction.events.head.getCreated.contractId
+  }
+
+  protected def create(
```

Comment:
This variant of `create` is added to avoid the Participant issuing back pressure signals during ACS setup with a ledger client.